LayoutTests/fullscreen/full-screen-keyboard-lock-option-enabled.html

```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>This tests that full screen mode with keyboard lock option allows ESC key event.</title>
+</head>
+<body>
+    <div width: 100%; height: 100%;" id="target">
+        <p>Test passes if ESC key received by the test case after enter to fullscreen.</p>
+    </div>
+    <script src="full-screen-test.js"></script>
+    <script>
+
+        window.internals.settings.setFullScreenKeyboardLock(true);
```

Comment:
This seems to be using old JS idioms and legacy fullscreen events. I haven't tried the below, but could you maybe simplify the tests by rewriting it in this style: 

```
window.internals.settings.setFullScreenKeyboardLock(true);
var event;
runWithKeyDown(async () => {
    const target = document.querySelector("#target");
    try {
        await target.requestFullscreen({ keyboardLock: "browser" });
        testExpected("document.fullscreenElement", true);
        event = await new Promise((resolve) => {
            waitForEvent(
                document,
                "keypress",
                resolve,
                /*endit=*/ false,
                /*once=*/ true
            );
            eventSender.keyDown("escape");
        });
        testExpected("event.keyCode", 27); // 'Escape'
    } catch (err) {
        logResult(
            false,
            `An error occurred while trying to enter full-screen mode: ${err.message}`
        );
    } finally {
        endTest();
    }
});
```

Reply:
I think there should even be some way of getting rid of that global `var event` and testing for the key code without relying on the global scope.

Reply:
@marcoscaceres  Thanks for the advice. Updated the test cases.